### PR TITLE
fix(remap): Fix to_syslog_* function names

### LINKED ISSUE #5959: rename `to_syslog_severity` to `parse_syslog_severity`
This one is incorrectly named. It behaves similar to other `parse_*` functions (take in a string, try to parse it to the relevant type, error if it can't), but is named as a `to_*` function.

### PR DESCRIPTION
**Pull request by lucperkins**
Changes to names of the `to_syslog_*` functions to `parse_syslog_*`, which better reflects the role of those functions in VRL given that `to_*` functions perform type coercion. It also adds some missing behavior tests for `parse_syslog_level` and `parse_syslog_severity`.

Apologies to @jbradleynh for already changing your handiwork 😄 This problem is on our side!

Closes #5959

### CODE REVIEW

File: docs/reference/remap/parse_syslog_level.cue [hunk 1]
```diff
@@ -18,14 +18,14 @@ remap: functions: to_syslog_level: {
 	examples: [
 		{
 			title: "Convert Syslog severity to level"
-			input: log: severity: "5"
-			source: ".log_level = to_syslog_level(.severity)"
+			input: log: severity: 5
+			source: ".level = to_syslog_level(.severity)"
```
Comment (FungusHumungus) [inline docs/reference/remap/parse_syslog_level.cue:22]:
```suggestion
			source: ".level = parse_syslog_level(.severity)"
```

File: docs/reference/remap/parse_syslog_level.cue [hunk 2]
```diff
@@ -18,14 +18,14 @@ remap: functions: to_syslog_level: {
...
 			output: input & {log: level: "notice"}
 		},
 		{
 			title: "Error"
-			input: log: severity: "1337"
-			source: ".log_level = to_syslog_level(.severity)"
+			input: log: severity: 1337
+			source: ".level = to_syslog_level(.severity)"
```
Comment (FungusHumungus) [inline docs/reference/remap/parse_syslog_level.cue:28]:
```suggestion
			source: ".level = parse_syslog_level(.severity)"
```

**Review by FungusHumungus — APPROVED** [review]

**Comment by binarylogic** [comment]
I'm a little confused with these names. What would the `parse_syslog_level` and `parse_syslog_severity` functions return? It seems odd that the `parse_syslog_level` would return an integer. You could make the case that it should return a map:

```
parse_syslog_level("info")
=> {"severity" => 2}
```

cc @JeanMertz on if we want to proceed with this change.

**Comment by JeanMertz** [comment]
Note that the original issue proposed renaming `to_syslog_severity` to `parse_syslog_severity` (it didn't discuss renaming `to_syslog_level`), since this function takes a string, and _parses_ it to a different type (in this case an integer), it can fail if parsing fails. This is similar to all our other parsing functions.

I don't think a defining characteristic of a parsing function is that it _must_ resolve to a map (and I'm not too sure what the value of that would be, since you already know the string value, since you just passed it into the function as its argument).

I _don't_ think we should rename `to_syslog_level`, since that takes an integer, not a string, and even though it can fail, that's something other `to_*` functions can do as well.

I admit the lines are thin between some `to_*` and `parse_*` functions, but so far, the biggest distinction has been that `parse_` takes (only!) a string as input and parses it to some other type, whereas `to_*` functions take several input types and convert it to some other type.

**Comment by lucperkins** [comment]
@JeanMertz I haven't touched this in a while and I'm completely agnostic on the naming. I feel like it should be your call on this. Let me know how you'd like us to proceed on this and I'll make it happen.

**Comment by JeanMertz** [comment]
I'll deref to @binarylogic. I'm happy to close this PR and ship with the existing function names. We can always revisit before Vector 1.0 if we think there are clear cases where the current names don't make sense.

**Comment by jszwedko** [comment]
Bump @binarylogic . Do you have any opinions here?

I'm inclined to go with what you originally suggested in https://github.com/timberio/vector/issues/5959#issuecomment-759576361 which means we can just close this.

**Comment by jszwedko** [comment]
I'm gonna close this in-lieu of #5959

**Pull request closed without merging**